src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -7,8 +7,32 @@
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
-    @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException(
+                    "Login cannot be null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException(
+                    "User with this login already exists");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException(
+                    "Login must be at least 6 characters long");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException(
+                    "Password must be at least 6 characters long");
+        }
+        if (user.getAge() < 18) {
+            throw new RegistrationException(
+                    "User must be at least 18 years old");
+        }
+
+        storageDao.add(user);
```

Comment:
You are adding the user to the storage, which is good. However, you are not checking if the add operation was successful. It's always a good idea to check the result of an operation and handle any errors that may occur.